cli/resources.rs

```diff
@@ -101,6 +101,13 @@ enum Repr {
   Worker(WorkerChannels),
 }
 
+fn not_found(rid: ResourceId) -> Error {
+  Error::new(
+    ErrorKind::NotFound,
+    format!("Resource with rid {:?} not found", rid),
+  )
+}
```

Comment:
Thanks, looks like [`from_raw_os_error`](https://doc.rust-lang.org/std/io/struct.Error.html#method.from_raw_os_error) is the tool for the job, because there's no error kind like that in `std::io::ErrorKind`.

Reply:
Use BadResource

https://github.com/denoland/deno/blob/ca000392857b4c79a3609ddbc20073222498998b/cli/msg.rs#L28

https://github.com/denoland/deno/blob/ca000392857b4c79a3609ddbc20073222498998b/cli/deno_error.rs#L49-L51

Reply:
@ry that's not possible... We need this error to be of `std::io::Error` type because `AsyncWrite` trait needs that, see https://github.com/denoland/deno/issues/2249 for longer description